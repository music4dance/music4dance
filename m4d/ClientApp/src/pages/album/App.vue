<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
