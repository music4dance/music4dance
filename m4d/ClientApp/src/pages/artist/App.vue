<script setup lang="ts">
import { ArtistModel } from "@/models/ArtistModel";
import { SongFilter } from "@/models/SongFilter";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { TypedJSON } from "typedjson";
import { Song } from "@/models/Song";
import { useSongSelector } from "@/composables/useSongSelector";

declare const model_: string;

const model = TypedJSON.parse(model_, ArtistModel)!;

const danceDB = safeDanceDatabase();
const { songs: selected, select: selectSong } = useSongSelector();

const title = `Artist: ${model.artist}`;
const filter = new SongFilter();
const histories = model.histories ?? [];
const songs = histories.map((h) => Song.fromHistory(h));
const danceIds = [
  ...songs.reduce((acc, h) => {
    const ratings = h.danceRatings;
    if (ratings) {
      ratings.forEach((r) => acc.add(r.danceId));
    }
    return acc;
  }, new Set<string>()),
];
const dances = danceIds.map((id) => danceDB.dances.find((d) => d.id === id)!);
</script>

<template>
  <PageFrame id="app" :title="title">
    <p>
      {{ histories.length }} {{ model.artist }} songs with suggestions for dances to dance to them:
      <span v-for="(dance, idx) in dances" :key="dance.id"
        ><span v-if="idx + 1 === dances.length && dances.length > 1"> and </span
        ><a :href="'/dances/' + dance.seoName">{{ dance.name }}</a
        ><span v-if="idx + 1 < dances.length && dances.length > 2">, </span></span
      >
    </p>
    <SongTable
      :histories="histories"
      :filter="filter"
      :hide-sort="true"
      :hidden-columns="['artist', 'length', 'track']"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
