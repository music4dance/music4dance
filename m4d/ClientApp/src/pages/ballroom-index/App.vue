<script setup lang="ts">
import BallroomList from "@/components/BallroomList.vue";
import BlogTagLink from "@/components/BlogTagLink.vue";
import CompetitionCategoryTable from "@/components/CompetitionCategoryTable.vue";
import PageFrame from "@/components/PageFrame.vue";
import TempiLink from "@/components/TempiLink.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";
import { CompetitionCategory, CompetitionGroup } from "@/models/Competition";
import { TypedJSON } from "typedjson";

declare const model_: string;
const group = TypedJSON.parse(model_, CompetitionGroup)!;
const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Ballroom", active: true }];

function categoryLink(category: CompetitionCategory): string {
  return `/dances/${category.canonicalName}`;
}
</script>

<template>
  <PageFrame id="app" title="Competition Ballroom Dancing" :breadcrumbs="breadcrumbs">
    <BallroomList>
      One of the
      <a
        href="https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/"
        target="_blank"
      >
        core ideas</a
      >
      behind <a href="https://www.music4dance.net">music4dance</a> is to find interesting music to
      build playlists for competitions rounds. This page is a central location to start a search for
      songs for your own playlists based on the tempo definitions of competition ballroom dance
      established by
      <a href="http://www.worlddancesport.org/Rule/Athlete/Competition" target="_blank"
        >World Dance Council</a
      >
      and
      <a href="https://www.ndca.org/pages/ndca_rule_book/Default.asp" target="_blank"
        >National Dance Council of America</a
      >
      and our database of songs.
    </BallroomList>
    <div v-for="category in group.categories" :key="category.name">
      <h2>
        <a :href="categoryLink(category)">{{ category.name }}</a>
      </h2>
      <CompetitionCategoryTable
        :dances="category.round"
        :title="category.fullRoundTitle"
        :use-full-name="false"
      ></CompetitionCategoryTable>
    </div>
    <div>
      <TempiLink />
      <BlogTagLink title="Ballroom" tag="ballroom"></BlogTagLink>
    </div>
  </PageFrame>
</template>
