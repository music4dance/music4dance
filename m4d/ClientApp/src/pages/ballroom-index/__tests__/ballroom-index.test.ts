import { describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestPageSnapshot";
import { model } from "./model";
import App from "../App.vue";

describe("CompetitionCategory", () => {
  test("Renders the Ballroom Index Page", () => {
    testPageSnapshot(App, model);
  });
});
