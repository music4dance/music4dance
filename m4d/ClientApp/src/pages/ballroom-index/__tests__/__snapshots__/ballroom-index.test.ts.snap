// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompetitionCategory > Renders the Ballroom Index Page 1`] = `
"<div><span context="[object Object]">MainMenu</span>
  <nav aria-label="breadcrumb">
    <nav aria-label="breadcrumb" style="padding: 0.25rem 0.5rem;">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a class="" href="/">music4dance</a></li>
        <li class="breadcrumb-item"><a class="" href="/dances">Dances</a></li>
        <li class="breadcrumb-item active"><span aria-current="location">Ballroom</span></li>
      </ol>
    </nav>
  </nav>
  <div id="body-content" class="container-fluid body-content">
    <h1>Competition Ballroom Dancing</h1>
    <div>
      <div>
        <p> One of the <a href="https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/" target="_blank"> core ideas</a> behind <a href="https://www.music4dance.net">music4dance</a> is to find interesting music to build playlists for competitions rounds. This page is a central location to start a search for songs for your own playlists based on the tempo definitions of competition ballroom dance established by <a href="http://www.worlddancesport.org/Rule/Athlete/Competition" target="_blank">World Dance Council</a> and <a href="https://www.ndca.org/pages/ndca_rule_book/Default.asp" target="_blank">National Dance Council of America</a> and our database of songs. </p>
        <ul style="list-style-type: disc;">
          <li> Click on any of the tempos in the table below (in the MPM, DanceSport, or NDCA columns) to go to a list of songs of that tempo </li>
          <li>Click on the dance names to go to our details information page for that dance style</li>
          <li> Read our <a href="https://music4dance.blog/im-a-ballroom-dancer-can-i-find-practice-songs-that-are-at-competition-tempo-revisited/" target="_blank"> blog post</a> for step by step instructions on using our catalog to find music of specific style and tempo. </li>
          <li><a href="/identity/account/register">Register</a> (it's free) to start building your own music lists. </li>
        </ul>
      </div>
      <div>
        <h2><a href="/dances/american-smooth">American Smooth</a></h2>
        <div>
          <h4>An American Smooth four dance round consists of the following:</h4>
          <div class="table-responsive">
            <table id="BootstrapVueNext__ID__v-0__" class="table b-table table-hover table-striped" aria-busy="false">
              <!---->
              <thead class="">
                <tr class="">
                  <th scope="col" class="">Name
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">Meter
                    <!---->
                  </th>
                </tr>
                <!---->
              </thead>
              <tbody class="">
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="slow-waltz">Slow Waltz</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90&amp;sortorder=Dances">28-30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90&amp;sortorder=Dances">84-90</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=90&amp;tempomax=90&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=90&amp;tempomax=90&amp;sortorder=Dances">90</a>
                  </td>
                  <td class="">
                    <!---->3/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="tango-(ballroom)">Tango (Ballroom)</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">120</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">120</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="slow-foxtrot">Slow Foxtrot</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128&amp;sortorder=Dances">30-32</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128&amp;sortorder=Dances">120-128</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">120</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="viennese-waltz">Viennese Waltz</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162&amp;sortorder=Dances">53-54</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162&amp;sortorder=Dances">159-162</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=159&amp;sortorder=Dances">53</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=159&amp;sortorder=Dances">159</a>
                  </td>
                  <td class="">
                    <!---->3/4
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
              <!---->
            </table>
          </div>
        </div>
      </div>
      <div>
        <h2><a href="/dances/international-standard">International Standard</a></h2>
        <div>
          <h4>An International Standard five dance round consists of the following:</h4>
          <div class="table-responsive">
            <table id="BootstrapVueNext__ID__v-3__" class="table b-table table-hover table-striped" aria-busy="false">
              <!---->
              <thead class="">
                <tr class="">
                  <th scope="col" class="">Name
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">Meter
                    <!---->
                  </th>
                </tr>
                <!---->
              </thead>
              <tbody class="">
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="slow-waltz">Slow Waltz</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90&amp;sortorder=Dances">28-30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90&amp;sortorder=Dances">84-90</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=87&amp;tempomax=87&amp;sortorder=Dances">29</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=87&amp;tempomax=87&amp;sortorder=Dances">87</a>
                  </td>
                  <td class="">
                    <!---->3/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="tango-(ballroom)">Tango (Ballroom)</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=128&amp;tempomax=128&amp;sortorder=Dances">32</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=128&amp;tempomax=128&amp;sortorder=Dances">128</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=124&amp;tempomax=132&amp;sortorder=Dances">31-33</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=TGO&amp;tempomin=124&amp;tempomax=132&amp;sortorder=Dances">124-132</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="viennese-waltz">Viennese Waltz</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=174&amp;tempomax=190&amp;sortorder=Dances">58-63.3</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=174&amp;tempomax=190&amp;sortorder=Dances">174-190</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=174&amp;tempomax=174&amp;sortorder=Dances">58</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=174&amp;tempomax=174&amp;sortorder=Dances">174</a>
                  </td>
                  <td class="">
                    <!---->3/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="slow-foxtrot">Slow Foxtrot</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=112&amp;tempomax=120&amp;sortorder=Dances">28-30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=112&amp;tempomax=120&amp;sortorder=Dances">112-120</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=112&amp;tempomax=112&amp;sortorder=Dances">28</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SFT&amp;tempomin=112&amp;tempomax=112&amp;sortorder=Dances">112</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="quickstep">Quickstep</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=QST&amp;tempomin=200&amp;tempomax=208&amp;sortorder=Dances">50-52</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=QST&amp;tempomin=200&amp;tempomax=208&amp;sortorder=Dances">200-208</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=QST&amp;tempomin=200&amp;tempomax=200&amp;sortorder=Dances">50</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=QST&amp;tempomin=200&amp;tempomax=200&amp;sortorder=Dances">200</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
              <!---->
            </table>
          </div>
        </div>
      </div>
      <div>
        <h2><a href="/dances/american-rhythm">American Rhythm</a></h2>
        <div>
          <h4>An American Rhythm five dance round consists of the following:</h4>
          <div class="table-responsive">
            <table id="BootstrapVueNext__ID__v-6__" class="table b-table table-hover table-striped" aria-busy="false">
              <!---->
              <thead class="">
                <tr class="">
                  <th scope="col" class="">Name
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">Meter
                    <!---->
                  </th>
                </tr>
                <!---->
              </thead>
              <tbody class="">
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="cha-cha">Cha Cha</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">120</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">30</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=120&amp;sortorder=Dances">120</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="rumba">Rumba</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=120&amp;tempomax=124&amp;sortorder=Dances">30-31</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=120&amp;tempomax=124&amp;sortorder=Dances">120-124</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=124&amp;tempomax=124&amp;sortorder=Dances">31</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=124&amp;tempomax=124&amp;sortorder=Dances">124</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="east-coast-swing">East Coast Swing</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=ECS&amp;tempomin=136&amp;tempomax=144&amp;sortorder=Dances">34-36</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=ECS&amp;tempomin=136&amp;tempomax=144&amp;sortorder=Dances">136-144</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=ECS&amp;tempomin=140&amp;tempomax=140&amp;sortorder=Dances">35</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=ECS&amp;tempomin=140&amp;tempomax=140&amp;sortorder=Dances">140</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="bolero">Bolero</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=BOL&amp;tempomin=96&amp;tempomax=104&amp;sortorder=Dances">24-26</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=BOL&amp;tempomin=96&amp;tempomax=104&amp;sortorder=Dances">96-104</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=BOL&amp;tempomin=96&amp;tempomax=96&amp;sortorder=Dances">24</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=BOL&amp;tempomin=96&amp;tempomax=96&amp;sortorder=Dances">96</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="mambo">Mambo</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=MBO&amp;tempomin=188&amp;tempomax=188&amp;sortorder=Dances">47</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=MBO&amp;tempomin=188&amp;tempomax=188&amp;sortorder=Dances">188</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=MBO&amp;tempomin=188&amp;tempomax=188&amp;sortorder=Dances">47</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=MBO&amp;tempomin=188&amp;tempomax=188&amp;sortorder=Dances">188</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
              <!---->
            </table>
          </div>
        </div>
      </div>
      <div>
        <h2><a href="/dances/international-latin">International Latin</a></h2>
        <div>
          <h4>An International Latin five dance round consists of the following:</h4>
          <div class="table-responsive">
            <table id="BootstrapVueNext__ID__v-9__" class="table b-table table-hover table-striped" aria-busy="false">
              <!---->
              <thead class="">
                <tr class="">
                  <th scope="col" class="">Name
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">DanceSport (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (MPM)
                    <!---->
                  </th>
                  <th scope="col" class="">NDCA (BPM)
                    <!---->
                  </th>
                  <th scope="col" class="">Meter
                    <!---->
                  </th>
                </tr>
                <!---->
              </thead>
              <tbody class="">
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="cha-cha">Cha Cha</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=128&amp;sortorder=Dances">30-32</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=120&amp;tempomax=128&amp;sortorder=Dances">120-128</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=124&amp;tempomax=124&amp;sortorder=Dances">31</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=CHA&amp;tempomin=124&amp;tempomax=124&amp;sortorder=Dances">124</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="samba">Samba</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SMB&amp;tempomin=100&amp;tempomax=104&amp;sortorder=Dances">50-52</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SMB&amp;tempomin=100&amp;tempomax=104&amp;sortorder=Dances">100-104</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SMB&amp;tempomin=100&amp;tempomax=100&amp;sortorder=Dances">50</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=SMB&amp;tempomin=100&amp;tempomax=100&amp;sortorder=Dances">100</a>
                  </td>
                  <td class="">
                    <!---->2/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="rumba">Rumba</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=100&amp;tempomax=108&amp;sortorder=Dances">25-27</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=100&amp;tempomax=108&amp;sortorder=Dances">100-108</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=100&amp;tempomax=100&amp;sortorder=Dances">25</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=RMB&amp;tempomin=100&amp;tempomax=100&amp;sortorder=Dances">100</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="paso-doble">Paso Doble</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=PDL&amp;tempomin=120&amp;tempomax=124&amp;sortorder=Dances">60-62</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=PDL&amp;tempomin=120&amp;tempomax=124&amp;sortorder=Dances">120-124</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=PDL&amp;tempomin=110&amp;tempomax=110&amp;sortorder=Dances">55</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=PDL&amp;tempomin=110&amp;tempomax=110&amp;sortorder=Dances">110</a>
                  </td>
                  <td class="">
                    <!---->2/4
                  </td>
                </tr>
                <!---->
                <tr class="">
                  <td class="">
                    <!----><a href="jive">Jive</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=JIV&amp;tempomin=168&amp;tempomax=176&amp;sortorder=Dances">42-44</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=JIV&amp;tempomin=168&amp;tempomax=176&amp;sortorder=Dances">168-176</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=JIV&amp;tempomin=172&amp;tempomax=172&amp;sortorder=Dances">43</a>
                  </td>
                  <td class="">
                    <!----><a href="/song/advancedsearch?dances=JIV&amp;tempomin=172&amp;tempomax=172&amp;sortorder=Dances">172</a>
                  </td>
                  <td class="">
                    <!---->4/4
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
              <!---->
            </table>
          </div>
        </div>
      </div>
      <div>
        <div>
          <div><img src="/images/icons/tools.png" alt="tools" width="24" height="24"> Explore relationships between the meter and tempi of different dance styles and find songs that match.</div>
          <div class="ms-3 my-1"><a class="btn btn-sm btn-tools" href="https://www.music4dance.net/home/tempi" role="button" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1.2em" height="1.2em">
                <path fill="currentColor" d="M10.804 8L5 4.633v6.734zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696z"></path>
              </svg>&nbsp;Try It!</a><a class="btn btn-sm btn-tools" href="https://music4dance.blog/music4dance-help/dance-tempi/" role="button" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1.2em" height="1.2em">
                <g fill="currentColor">
                  <path d="M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5M5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1z"></path>
                  <path d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1"></path>
                </g>
              </svg>&nbsp;Documentation</a><a class="btn btn-sm btn-tools" href="https://music4dance.blog/tag/tempo/" role="button" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1.2em" height="1.2em">
                <g fill="currentColor">
                  <path d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456l-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z"></path>
                  <path fill-rule="evenodd" d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5z"></path>
                </g>
              </svg>&nbsp;Blog Posts</a></div>
          <!--v-if-->
        </div>
        <div>
          <dd><img src="/vclient/src/assets/images/icons/info.png" alt="tools" width="24" height="24"> Read the <a href="https://music4dance.blog">music4dance blog</a> entries about <a href="https://music4dance.blog/tag/ballroom">Ballroom</a>. </dd>
        </div>
      </div>
    </div>
  </div>
  <div id="footer-content">
    <hr>
    <footer>
      <p> © 2025 - <a href="https://www.music4dance.net">Music4Dance.net</a> - <a href="https://www.music4dance.net/home/sitemap">Site Map</a> - <a href="https://www.music4dance.net/home/termsofservice">Terms of Service</a> - <a href="https://www.music4dance.net/home/privacypolicy">Privacy Policy</a> - <a href="https://www.music4dance.net/home/credits">Credits</a> - <a href="https://github.com/music4dance/music4dance" target="_blank">Code</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
